plutus-contract/src/Plutus/Contract/Test.hs

```diff
@@ -572,11 +573,12 @@ walletFundsExactChange :: Wallet -> Value -> TracePredicate
 walletFundsExactChange = walletFundsChangeImpl True
 
 walletFundsChangeImpl :: Bool -> Wallet -> Value -> TracePredicate
-walletFundsChangeImpl exact w dlt = TracePredicate $
-    flip postMapM (L.generalize $ (,) <$> Folds.walletFunds w <*> Folds.walletFees w) $ \(finalValue', fees) -> do
+walletFundsChangeImpl exact w dlt' = TracePredicate $
+    flip postMapM (L.generalize $ (,,) <$> Folds.walletFunds w <*> Folds.walletFees w <*> Folds.walletAdjustedTxEvents) $ \(finalValue', fees, txOutCosts) -> do
```

Comment:
For symmetry it would be better to let `Folds.walletAdjustedTxEvents` also only return the events for a given wallet. Then you don't have to filter in `calculateDelta`.

Reply:
As we found out `walletEvent` and `walletEvent'` return a bit different results. I renamed `walletAdjustedTxEvents` to `walletsAdjustedTxEvents` and added `walletAdjustedTxEvents :: Wallet -> ...`.

Reply:
I don't think this is the right conclusion to get from this. `walletEvent` and `walletEvent'` differ only in filtering out events for the given wallet. (look at the implementation, it's really simple) The issue lay in how you used it.

When I suggested the above I didn't realise you also needed the events from all the wallets. So instead of having 2 functions it should also be possible to fix the implementation of `walletsAdjustedTxEvents`.

Reply:
>The issue lay in how you used it. 

I agree, my problem is here I don't understand yet what's the issue. `walletAdjustedTxEvents` and `walletsAdjustedTxEvents` looks very similar and I don't get why they have a different behaviour.